### PR TITLE
Feat/use carousel

### PR DESCRIPTION
**Pull request by siddhantk232**
This **PR** includes:

- `useCarousel` hook.
-  an example implementing this hook.

### CODE REVIEW

File: src/hooks/useCarousel.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+   const setActive = index => {
+      if (index > data.length - 1 || index < 0) return
+
+      setCurrent(index)
+   }
+
+   const next = () => {
+      if (current === data.length - 1) setCurrent(0)
+      else setCurrent(current + 1)
```
Comment (prvnbist) [inline src/hooks/useCarousel.js:19]:
```suggestion
     (current + 1 + data.length) % data.length
```

File: src/hooks/useCarousel.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+   }
+
+   const previous = () => {
+      if (current === 0) setCurrent(data.length - 1)
+      else setCurrent(current - 1)
```
Comment (prvnbist) [inline src/hooks/useCarousel.js:24]:
```suggestion
      (current - 1 + data.length) % data.length
```

File: src/hooks/useCarousel.js (hunk 1 above)
Comment (siddhantk232) [reply src/hooks/useCarousel.js:19]: both are doing the same thing I guess as your code is hard for me to understand here.

File: src/hooks/useCarousel.js (hunk 2 above)
Comment (siddhantk232) [reply src/hooks/useCarousel.js:24]: if you really want I can commit this suggestion.

**Pull request closed without merging**